src/libraries/System.Private.CoreLib/src/System/AppDomain.cs

```diff
@@ -404,34 +403,38 @@ public void SetThreadPrincipal(IPrincipal principal)
                 switch (_principalPolicy)
                 {
                     case PrincipalPolicy.UnauthenticatedPrincipal:
-                        if (s_getUnauthenticatedPrincipal == null)
-                        {
-                            Type type = Type.GetType("System.Security.Principal.GenericPrincipal, System.Security.Claims", throwOnError: true)!;
-                            MethodInfo? mi = type.GetMethod("GetDefaultInstance", BindingFlags.NonPublic | BindingFlags.Static);
-                            Debug.Assert(mi != null);
-                            // Don't throw PNSE if null like for WindowsPrincipal as UnauthenticatedPrincipal should
-                            // be available on all platforms.
-                            s_getUnauthenticatedPrincipal = mi.CreateDelegate<Func<IPrincipal>>();
-                        }
-
-                        principal = s_getUnauthenticatedPrincipal();
+                        principal = (IPrincipal?)GetDefaultPrincipal(null);
```

Comment:
```suggestion
                        principal = (IPrincipal)GetDefaultPrincipal(null);
```

This will never return a null value.